### PR TITLE
ar71xx: add support for TP-LINK TL-WR941N V7

### PR DESCRIPTION
**Pull request by imbrolla**
Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

### CODE REVIEW

File: target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr941n-v7.c [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	phy_address = (0x17 & ((addr_temp >> 4) | 0x10));
+	reg_address = ((addr_temp << 1) & 0x1e);
+	s27_rd_csr_low = (unsigned int) phy_reg_read(phy_address, reg_address);
+
+	reg_address = reg_address | 0x1;
+	s27_rd_csr_high = (unsigned int) phy_reg_read(phy_address, reg_address);
+	s27_rd_csr = (s27_rd_csr_high << 16) | s27_rd_csr_low ;
+	
+	return (s27_rd_csr);
+}
```
Comment (981213) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr941n-v7.c]: Have you noticed that here are some code doing mdio operations  by writing registers directly when you copy code from other people? That's the reason why the original author of this code (hackpascal) thinks that this device support won't be acceptable by openwrt.

File: target/linux/ar71xx/base-files/etc/diag.sh [hunk 2]
```diff
@@ -98,6 +98,7 @@ get_status_led() {
 	tl-wr810n|\
 	tl-wr810n-v2|\
 	tl-wr940n-v4|\
+	tl-wr941n-v7|\
```
Comment (981213) [inline target/linux/ar71xx/base-files/etc/diag.sh]: That's a green led, not a blue one according to mach code.

File: target/linux/ar71xx/config-4.14 [hunk 3]
```diff
@@ -239,6 +239,7 @@ CONFIG_ATH79=y
 # CONFIG_ATH79_MACH_TL_WR940N_V4 is not set
 # CONFIG_ATH79_MACH_TL_WR941ND is not set
 # CONFIG_ATH79_MACH_TL_WR941ND_V6 is not set
+CONFIG_ATH79_MACH_TL_WR941N_V7=y
```
Comment (981213) [inline target/linux/ar71xx/config-4.14]: Wrong place to put this. you should put this to tiny/config-default

**Review by 981213 — COMMENTED** [review]
It's for ar71xx not ath79, and you should squash all your commits into one and follow https://openwrt.org/submitting-patches to add correct title/description/signoff etc.
And I don't think this device support you copied from hackpascal will be accepted due to the switch reset operations directly written in mach files. And I don't think ar71xx currently accept any new device support at all.

**Comment by imbrolla** [comment]
1

**Pull request closed without merging**